azuremetadata

```diff
@@ -44,36 +46,51 @@ try:
 
     util = azuremetadatautils.AzureMetadataUtils(data)
 
-    # Metadata API on ASM (Azure Classic) instances does not produce some of the data
-    # required for registration to Public Cloud update infrastructure.
-    # If the metadata utility was called with arguments used by cloud-regionsrv-client,
-    # we have to ensure that those parameters are defined and not raising QueryException errors.
+    # Metadata API on ASM (Azure Classic) instances does not produce some of
+    # the data required for registration to Public Cloud update infrastructure.
+    # If the metadata utility was called with arguments used by
+    # cloud-regionsrv-client, we have to ensure that those parameters are
+    # defined and not raising QueryException errors.
     if (
         util.get_regionsrv_client_data_provider() == " ".join(sys.argv)
     ):
         # generate subscriptionId if it's unavailable
         if 'subscriptionId' not in util.available_params:
-            if len(util.available_params) > 1:
-                # if API response succeeded, but no subscriptionId is defined -- generate random ID
-                data['subscriptionId'] = 'classic-{}'.format(random.randint(0, 1e9))
+            # try the kvp data or generate random ID
+            if os.path.exists('/var/lib/hyperv/.kvp_pool_3'):
+                kvp_key_size = 512
+                kvp_value_size = 2048
+                kvp_pool_3 = open('/var/lib/hyperv/.kvp_pool_3', 'rb')
```

Comment:
It would be better to add `with open...as `
Also, the file path could go in a variable as it is used a couple of times.